src/Catalog/DeleteCatalogItem.cs

```diff
@@ -0,0 +1,121 @@
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.Type), catalogEntry.CreateUriNode(Schema.DataTypes.Permalink));
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.CatalogTimeStamp), catalogEntry.CreateLiteralNode(TimeStamp.ToString("O"), Schema.DataTypes.DateTime));
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.CatalogCommitId), catalogEntry.CreateLiteralNode(CommitId.ToString()));
+
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.Published), catalogEntry.CreateLiteralNode(_published.ToString("O"), Schema.DataTypes.DateTime));
+
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.Id), catalogEntry.CreateLiteralNode(_id));
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.OriginalId), catalogEntry.CreateLiteralNode(_id));
+                catalogEntry.Assert(catalogEntrySubject, catalogEntry.CreateUriNode(Schema.Predicates.Version), catalogEntry.CreateLiteralNode(_version));
+
+                SetIdVersionFromGraph(catalogEntry);
+
+                //  create JSON content
+                JObject frame = context.GetJsonLdContext("context.Catalog.json", GetItemType());
+                StorageContent content = new StringStorageContent(Utils.CreateArrangedJson(catalogEntry, frame), "application/json", "no-store");
+               
+                return content;
+            }
+        }
+
+        public override IGraph CreatePageContent(CatalogContext context)
+        {
+            Uri resourceUri = new Uri(GetBaseAddress() + GetRelativeAddress());
+
+            Graph graph = new Graph();
+
+            INode subject = graph.CreateUriNode(resourceUri);
+
+            INode idPredicate = graph.CreateUriNode(Schema.Predicates.Id);
+            INode versionPredicate = graph.CreateUriNode(Schema.Predicates.Version);
+
+            if (_id != null)
+            {
+                graph.Assert(subject, idPredicate, graph.CreateLiteralNode(_id));
+            }
+
+            if (_version != null)
+            {
+                graph.Assert(subject, versionPredicate, graph.CreateLiteralNode(_version));
+            }
+
+            return graph;
+        }
+
+        void SetIdVersionFromGraph(IGraph graph)
+        {
+            INode idPredicate = graph.CreateUriNode(Schema.Predicates.Id);
+            INode versionPredicate = graph.CreateUriNode(Schema.Predicates.Version);
+
+            INode rdfTypePredicate = graph.CreateUriNode(Schema.Predicates.Type);
+            Triple resource = graph.GetTriplesWithPredicateObject(rdfTypePredicate, graph.CreateUriNode(GetItemType())).First();
```

Comment:
are we guaranteed to get a value, why first vs. single? Can you add some comments.